src/platform/mock-local-home-platform.ts

```diff
@@ -1,144 +1,59 @@
+  /**
+   * Constructs a new MockLocalHomePlatform instance using an App instance
+   * @param app the AppStub that acts as an interface for intent handlers
+   */
+  public constructor(app: AppStub) {
     this.app = app;
   }
 
-  private onNewDeviceIdRegistered(localDeviceId: string) {
-    this.newDeviceRegisteredActions.forEach(newDeviceRegisteredAction => {
-      newDeviceRegisteredAction(localDeviceId);
-    });
-  }
-
-  public async getNextDeviceIdRegistered(): Promise<string> {
-    return new Promise(resolve => {
-      this.newDeviceRegisteredActions.push(localDeviceId => {
-        resolve(localDeviceId);
-      });
-    });
-  }
-
-  public isHomeAppReady(): boolean {
-    return this.homeAppReady;
-  }
-
-  public notifyHomeAppReady(): void {
-    this.homeAppReady = true;
-  }
-
-  //  Singleton getter
-  public static getInstance(): MockLocalHomePlatform {
-    if (!MockLocalHomePlatform.instance) {
-      MockLocalHomePlatform.instance = new MockLocalHomePlatform();
-    }
-    return MockLocalHomePlatform.instance;
-  }
-
   public getDeviceManager(): smarthome.DeviceManager {
     return this.deviceManager;
   }
 
-  public getLocalDeviceIdMap(): Map<string, string> {
-    return this.localDeviceIds;
-  }
-
-  private setupUDP() {
-    this.udpScanConfigs.forEach(udpScanConfig => {
-      this.mockNetwork!.registerUDPListener(
-        this,
-        udpScanConfig.listenPort,
-        udpScanConfig.broadcastAddress
-      );
-    });
+  public isDeviceIdRegistered(deviceId: string): boolean {
+    return this.localDeviceIds.has(deviceId);
   }
 
-  private sendUDPBroadcast(scanConfig: UDPScanConfig) {
-    const packetBuffer = Buffer.from(scanConfig.discoveryPacket, 'hex');
-    this.mockNetwork!.sendUDPMessage(
-      packetBuffer,
-      scanConfig.broadcastPort,
-      scanConfig.broadcastAddress,
-      scanConfig.listenPort,
-      scanConfig.broadcastAddress
-    );
+  public getLocalDeviceId(deviceId: string): string {
+    if (!this.isDeviceIdRegistered(deviceId)) {
+      throw new Error(ERROR_NO_LOCAL_DEVICE_ID_FOUND);
+    }
+    return this.localDeviceIds.get(deviceId)!;
   }
 
-  // Establish fulfillment path using app code
-  async onUDPMessage(msg: Buffer, rinfo: RemoteAddressInfo): Promise<void> {
-    if (this.app === undefined) {
-      throw new Error('Cannot trigger IdentifyRequest: App was undefined');
+  /**
+   * Takes a `discoveryBuffer` and passes it to the fulfillment app in an `IdentifyRequest`
+   * @param discoveryBuffer  The buffer to be included in the `IdentifyRequest` scan data
+   * @returns  The next localDeviceId registered to the Local Home Platform
+   */
+  public async triggerIdentify(discoveryBuffer: Buffer): Promise<string> {
+    console.debug('Received discovery payload:', discoveryBuffer);
+
+    // Cannot start processing until all handlers have been set on the `App`
+    if (!this.app.identifyHandler) {
+      return Promise.reject(new Error(ERROR_UNDEFINED_IDENTIFY_HANDLER));
```

Comment:
You don't need to wrap values in rejects or resolves. Running `await` on any function that doesn't return a Promise will work fine. `await (1+1)` returns 2 without having to be explicit about it being a promise.

So here you can just `throw new Error(ERROR_UNDEFINED_IDENTITY_HANDLER)`

and you can just `return device.verificationId` without wrapping.

Reply:
Done and noted.